### ISSUE TITLE
Cannot register 2 metrics with the same name: /tensorflow/api/keras/optimizers

### ISSUE THREAD

**Issue by karanveersingh5623** [tensorflow/tensorflow#52951]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): 
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): Docker image 
- TensorFlow version: 2.6.1
- Keras version: Getting the above error when import
- Python version: Python 3.6
- Installed using virtualenv? pip? conda?: pip (Docker image)
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 11.4
- GPU model and memory: Nvidia A100 40GB 

**Describe the problem**
Looks like this issue , https://github.com/keras-team/keras/issues/15579
Is it the right one ? because after applying the fix mentioned in above link "pip install tensorflow==2.6.2" , getting below error
```
root@6a893d98bbb1:/app/project# python3 download_process.py --help
WARNING:tensorflow:From /usr/local/lib/python3.6/dist-packages/tensorflow/python/compat/v2_compat.py:101: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
Instructions for updating:
non-resource variables are not supported in the long term
Traceback (most recent call last):
  File "download_process.py", line 13, in <module>
    from utils import *
  File "/app/project/utils.py", line 4, in <module>
    from object_detection.inputs import train_input
  File "/usr/local/lib/python3.6/dist-packages/object_detection/inputs.py", line 26, in <module>
    from object_detection.builders import model_builder
  File "/usr/local/lib/python3.6/dist-packages/object_detection/builders/model_builder.py", line 29, in <module>
    from object_detection.builders import matcher_builder
  File "/usr/local/lib/python3.6/dist-packages/object_detection/builders/matcher_builder.py", line 23, in <module>
    from object_detection.matchers import bipartite_matcher  # pylint: disable=g-import-not-at-top
  File "/usr/local/lib/python3.6/dist-packages/object_detection/matchers/bipartite_matcher.py", line 20, in <module>
    from tensorflow.contrib.image.python.ops import image_ops
ModuleNotFoundError: No module named 'tensorflow.contrib'
```

**Provide the exact sequence of commands / steps that you executed before running into the problem**

- Build Image by running below dockerfile

```
FROM tensorflow/tensorflow:2.3.0-gpu

RUN apt-get update --fix-missing && apt-get install -y \
    ffmpeg \
    git \
    git-core \
    g++ \
    pkg-config \
    python3-pip \
    unzip \
    vim \
    wget \
    zip \
    zlib1g-dev

WORKDIR /app

COPY requirements.txt .
RUN pip3 install -r requirements.txt
RUN pip3 install git+https://github.com/philferriere/cocoapi.git#subdirectory=PythonAPI

ENV TF_CPP_MIN_LOG_LEVEL=2

RUN wget https://github.com/protocolbuffers/protobuf/releases/download/v3.13.0/protoc-3.13.0-linux-x86_64.zip && \
    unzip protoc-3.13.0-linux-x86_64.zip -d /app/protobuf/

ENV PATH "$PATH:/app/protobuf/bin"

RUN git clone https://github.com/tensorflow/models.git && \
    cd /app/models/research/ && \
    protoc object_detection/protos/*.proto --python_out=. && \
    cp object_detection/packages/tf2/setup.py . &&\
    python -m pip install .
```

- Spawned container and inside it trying to run a pre-processing script. Below are packages used in it
   
```
import argparse
import io
import os
import subprocess

import ray
import tensorflow.compat.v1 as tf
tf.disable_v2_behavior()
#import tensorflow as tf
from PIL import Image
from psutil import cpu_count
tf.disable_v2_behavior()
from utils import *
from object_detection.utils import dataset_util, label_map_util
```

- Error Log Output
```
2021-11-04 10:25:18.325660: E tensorflow/core/lib/monitoring/collection_registry.cc:77] Cannot register 2 metrics with the same name: /tensorflow/api/keras/optimizers
Traceback (most recent call last):
  File "download_process.py", line 13, in <module>
    from utils import *
  File "/app/project/utils.py", line 4, in <module>
    from object_detection.inputs import train_input
  File "/usr/local/lib/python3.6/dist-packages/object_detection/inputs.py", line 26, in <module>
    from object_detection.builders import model_builder
  File "/usr/local/lib/python3.6/dist-packages/object_detection/builders/model_builder.py", line 25, in <module>
    from object_detection.builders import box_predictor_builder
  File "/usr/local/lib/python3.6/dist-packages/object_detection/builders/box_predictor_builder.py", line 20, in <module>
    from object_detection.predictors import convolutional_box_predictor
  File "/usr/local/lib/python3.6/dist-packages/object_detection/predictors/convolutional_box_predictor.py", line 26, in <module>
    from object_detection.core import box_predictor
  File "/usr/local/lib/python3.6/dist-packages/object_detection/core/box_predictor.py", line 137, in <module>
    class KerasBoxPredictor(tf.keras.layers.Layer):
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/util/lazy_loader.py", line 62, in __getattr__
    module = self._load()
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/util/lazy_loader.py", line 45, in _load
    module = importlib.import_module(self.__name__)
  File "/usr/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/usr/local/lib/python3.6/dist-packages/keras/__init__.py", line 25, in <module>
    from keras import models
  File "/usr/local/lib/python3.6/dist-packages/keras/models.py", line 20, in <module>
    from keras import metrics as metrics_module
  File "/usr/local/lib/python3.6/dist-packages/keras/metrics.py", line 26, in <module>
    from keras import activations
  File "/usr/local/lib/python3.6/dist-packages/keras/activations.py", line 20, in <module>
    from keras.layers import advanced_activations
  File "/usr/local/lib/python3.6/dist-packages/keras/layers/__init__.py", line 23, in <module>
    from keras.engine.input_layer import Input
  File "/usr/local/lib/python3.6/dist-packages/keras/engine/input_layer.py", line 21, in <module>
    from keras.engine import base_layer
  File "/usr/local/lib/python3.6/dist-packages/keras/engine/base_layer.py", line 43, in <module>
    from keras.mixed_precision import loss_scale_optimizer
  File "/usr/local/lib/python3.6/dist-packages/keras/mixed_precision/loss_scale_optimizer.py", line 18, in <module>
    from keras import optimizers
  File "/usr/local/lib/python3.6/dist-packages/keras/optimizers.py", line 26, in <module>
    from keras.optimizer_v2 import adadelta as adadelta_v2
  File "/usr/local/lib/python3.6/dist-packages/keras/optimizer_v2/adadelta.py", line 22, in <module>
    from keras.optimizer_v2 import optimizer_v2
  File "/usr/local/lib/python3.6/dist-packages/keras/optimizer_v2/optimizer_v2.py", line 37, in <module>
    "/tensorflow/api/keras/optimizers", "keras optimizer usage", "method")
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/monitoring.py", line 361, in __init__
    len(labels), name, description, *labels)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/monitoring.py", line 135, in __init__
    self._metric = self._metric_methods[self._label_length].create(*args)
tensorflow.python.framework.errors_impl.AlreadyExistsError: Another metric with the same name already exists.

```

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by SuyunCDN** [comment]
GREETINGS IT SPECIALIST! SUYUN CDN IS HIRING NOW! 

Company name: SuyunCDN

Title of position: Python Developer

Position type: Full Time, Part Time, or Contractual

Pay range: $800 - $2000


Location: Remote

Preferred Qualifications: 
	
●	Graduated in any IT related course
●	Proficient in English and Mandarin Language
●	Proficient in Python Coding

Description of responsibilities: 

●	Writing and testing code, debugging programs and integrating applications with third-party web services

●	Develop back-end components to improve responsiveness and overall performance
●	Integrate user-facing elements into applications
●	Test and debug programs
●	Improve functionality of existing systems
●	Implement security and data protection solutions
●	Assess and prioritize feature requests
●	Coordinate with internal teams to understand user requirements and provide technical solutions


Required experience: 
●	Work experience as a Python Developer

Required skills: 
●	Proficient with Django Python framework
●	Proficient with DevOps and Ansible
●	Extensive Knowledge on object-relational mapping (ORM)
●	Problem Solving Skill- Advance
●	Communication Skill - Advance
●	Highly Resourceful
●	Team-oriented personality


For information on SuyunCDN, including more information on our staff and our company culture, you may check our website at https://www.suyuncdn.com.

If interested, email your resume to suyuncdnclaudie@gmail.com
with the subject line of the job, you're applying for.
Check out our FB page for more info on available jobs we have
Suyuncdn

**Comment by mihaimaruseac** [comment]
This should be fixed by TF 2.6.2 or TF 2.7.0
